front/src/applications/stdcm/components/StdcmForm/StopDurationInput.tsx

```diff
@@ -0,0 +1,79 @@
+import { useMemo, useEffect, useState } from 'react';
+
+import { Input } from '@osrd-project/ui-core';
+import type { Status } from '@osrd-project/ui-core/dist/components/inputs/StatusMessage';
+import { useTranslation } from 'react-i18next';
+
+import { useOsrdConfActions } from 'common/osrdContext';
+import type { StdcmConfSliceActions } from 'reducers/osrdconf/stdcmConf';
+import type { StdcmPathStep } from 'reducers/osrdconf/types';
+import { useAppDispatch } from 'store';
+import { useDebounce } from 'utils/helpers';
+
+import { StdcmStopTypes } from '../../types';
+
+type StopDurationInputProps = {
+  pathStep: Extract<StdcmPathStep, { isVia: true }>;
+};
+
+const StopDurationInput = ({ pathStep }: StopDurationInputProps) => {
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation('stdcm');
+
+  const { updateStdcmPathStep } = useOsrdConfActions() as StdcmConfSliceActions;
+
+  const [stopDuration, setStopDuration] = useState('');
+  const debouncedStopDuration = useDebounce(stopDuration, 300);
+
+  const stopWarning = useMemo(
+    () =>
+      pathStep.stopType === StdcmStopTypes.DRIVER_SWITCH &&
+      pathStep.stopFor !== undefined &&
+      pathStep.stopFor < 3
+        ? {
+            status: 'warning' as Status,
+            message: t('trainPath.warningMinStopTime'),
+          }
+        : undefined,
+    [pathStep.stopType, pathStep.stopFor]
+  );
+  useEffect(() => {
+    setStopDuration(pathStep.stopFor !== undefined ? `${pathStep.stopFor}` : '');
+  }, [pathStep.stopFor]);
+
+  useEffect(() => {
+    const newStopDuration = debouncedStopDuration ? Number(debouncedStopDuration) : undefined;
+    if (newStopDuration !== pathStep.stopFor) {
+      dispatch(
+        updateStdcmPathStep({
+          id: pathStep.id,
+          updates: { stopFor: newStopDuration },
+        })
+      );
+    }
+  }, [debouncedStopDuration]);
+
+  if (pathStep.stopType === StdcmStopTypes.PASSAGE_TIME) {
+    return null;
+  }
```

Comment:
I think it would be better to don't let the component decide to render or not but keep it in the parent to prevent unnecessary mounting

Reply:
Done !